tests/unit/model/native-access-test.js

```diff
@@ -122,124 +123,135 @@ if (CUSTOM_MODEL_CLASS && HAS_NATIVE_PROXY) {
-      assert.equal(book.name, 'Temp title', 'native property access write changes the value');
-      assert.throws(
-        () => {
-          book.title;
-        },
-        /You attempted to access the `title` property/,
-        'Native access of an m3 property errors out'
-      );
+          /You attempted to access the `title` property/,
+          'Native access of an m3 property errors out'
+        );
+      }
     });
 
-    test('useNativeProperties can be set for each model type', function (assert) {
-      let useNativePropertiesCount = 0;
-      class DynamicSchema extends NativeSchema {
-        useNativeProperties(modelName) {
-          if (useNativePropertiesCount === 0) {
-            assert.equal(
-              modelName,
-              'com.example.bookstore.book',
-              'Passsed in the correct modelName to useNativeProperties hook'
-            );
-            useNativePropertiesCount++;
-            return true;
+    if (DEBUG) {
+      test('setting native properties deprecates when useNativeProperties is false', function (assert) {
+        class NonNativeSetSchema extends TestSchema {
+          setAttribute(modelName, attr, value, schemaInterface) {
+            assert.ok(false, 'should not call setAttribute when useNativeProperties is not true');
+            schemaInterface.setAttr('name', value);
           }
-
-          if (useNativePropertiesCount === 1) {
-            assert.equal(
-              modelName,
-              'com.example.bookstore.author',
-              'Passsed in the correct modelName to useNativeProperties hook'
-            );
-            useNativePropertiesCount++;
+          useNativeProperties() {
             return false;
           }
-          assert.ok(false, 'Should not reach useNativeProperties more than once per model');
         }
-      }
-      this.owner.register('service:m3-schema', DynamicSchema);
-
-      this.store.push({
-        data: {
-          id: 'urn:li:book:1',
-          type: 'com.example.bookstore.Book',
-          attributes: {
-            title: 'How to Win Friends and Influence People',
+        this.owner.register('service:m3-schema', NonNativeSetSchema);
+        this.store.push({
+          data: {
+            id: 'urn:li:book:1',
+            type: 'com.example.bookstore.Book',
+            attributes: {
+              title: 'How to Win Friends and Influence People',
+            },
           },
-        },
-      });
+        });
 
-      let book = this.store.peekRecord('com.example.bookstore.Book', 'urn:li:book:1');
+        let book = this.store.peekRecord('com.example.bookstore.Book', 'urn:li:book:1');
 
-      assert.equal(
-        book.title,
-        'How to Win Friends and Influence People',
-        'accessed field via native property get'
-      );
+        assert.expectDeprecation(() => {
+          book.name = 'Temp title';
+        }, `You set the property 'name' on a 'com.example.bookstore.book' with id 'urn:li:book:1'`);
```

Comment:
why don't we `if(DEBUG)` this assertion rather than the whole test?  the bulk of  the test still seems relevant in prod mode or am i missing something?

Reply:
Yeah good point, we would still want it to work in prod and I'm not sure we have a different test for it. I'll rename the test as well, I added the guard because the name suggested it was asserting the deprecation

Reply:
Done